src/inc/sift-events/sift-events.php

```diff
@@ -674,22 +739,39 @@ public static function update_or_create_order( string $order_id, \WC_Order $orde
 	 * @return void
 	 */
 	public static function transaction( \WC_Order $order, string $status, string $transaction_type ) {
-
-		if ( ! Sift_Event_Types::can_event_be_sent( Sift_Event_Types::$transaction ) ) {
+		// $transaction requires a positive amount, so we don't send it for free orders.
+		if ( self::is_free_order( $order ) || ! Sift_Event_Types::can_event_be_sent( Sift_Event_Types::$transaction ) ) {
 			return;
 		}
 
+		// Determine user and session context.
+		$user_id    = wp_get_current_user()->ID ?? null; // Check first for logged-in user.
```

Comment:
This is a big change from getting the user ID form the order... I think the idea here was `transaction()` is triggered from the Stripe webhooks, and I don't think the right user is available in this context.
```suggestion
		$user_id    = $order->get_user_id();
```

Reply:
This is getting undone...we don't even need the transaction event for $0 orders.